sdk/core/azure-core/src/main/java/com/azure/core/util/ClientOptions.java

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Client Options for setting common properties for example applicationId. Most of these properties are applied on
+ * request being send to Azure Service but some could be used for other purpose also for example applicationId could be
+ * used for telemetry.
+ */
+@Fluent
+public final class ClientOptions {
+    private final ClientLogger logger = new ClientLogger(ClientOptions.class);
+    private final Map<String, Header> headers = new ConcurrentHashMap<>();
```

Comment:
Does this need to be a concurrent hash map.

Reply:
The map doesn't need to exist at all. We just have setters and getters for headers and a property of type `Iterable<Header>`. Don't see the need to use the map here.